ninja format breaks compile of self driving server

# `ninja format` breaks compile of self driving server

When you run `ninja format`, it changes `src/execution/sql/aggregation_hash_table.cpp`, `src/execution/sql/join_hash_table.cpp`, `src/include/execution/sql/vector_projection_iterator.h`, `src/self_driving/model_server/model_server_manager.cpp`. The code it produces won't compile:

```
/Applications/CLion.app/Contents/bin/cmake/mac/bin/cmake --build /Users/ejeppinger/CLionProjects/terrier/cmake-build-debug --target noisepage
[0/2] Re-checking globbed directories...
[2/34] Building CXX object CMakeFiles/noisepage_objlib.dir/Unity/unity_34_cxx.cxx.o
FAILED: CMakeFiles/noisepage_objlib.dir/Unity/unity_34_cxx.cxx.o 
/usr/local/Cellar/llvm@8/8.0.1_3/bin/clang++  -DGFLAGS_IS_A_DLL=0 -DNOISEPAGE_USE_LOGGING -I../src/include -I_deps/src/count/count/../include -I_deps/src/cppzmq -I_deps/build/gflags/include -I_deps/src/ips4o/ips4o -I_deps/src/ips4o -I_deps/src/madoka -I_deps/src/nlohmann_json/include -I_deps/src/xbyak -isystem /usr/local/Cellar/llvm@8/8.0.1_3/include -isystem /usr/local/include -isystem _deps/src -isystem _deps/src/spdlog/include -isystem _deps/src/fast_float/include -isystem /usr/local/Cellar/zeromq/4.3.3/include -isystem ../third_party/libpg_query/../src/postgres/include -isystem ../third_party/libpg_query/.. -g -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -fPIC   -Werror -Wall -march=native -mcx16 -ggdb -O0 -fno-omit-frame-pointer -fno-optimize-sibling-calls -Wno-braced-scalar-init -fvisibility=hidden -fsanitize-blacklist=/Users/ejeppinger/CLionProjects/terrier/build-support/data/asan_sanitize_blocklist.txt -fcolor-diagnostics -std=c++17 -MD -MT CMakeFiles/noisepage_objlib.dir/Unity/unity_34_cxx.cxx.o -MF CMakeFiles/noisepage_objlib.dir/Unity/unity_34_cxx.cxx.o.d -o CMakeFiles/noisepage_objlib.dir/Unity/unity_34_cxx.cxx.o -c CMakeFiles/noisepage_objlib.dir/Unity/unity_34_cxx.cxx
In file included from CMakeFiles/noisepage_objlib.dir/Unity/unity_34_cxx.cxx:7:
In file included from /Users/ejeppinger/CLionProjects/terrier/src/self_driving/model_server/model_server_manager.cpp:1:
../src/include/self_driving/model_server/model_server_manager.h:150:35: error: no member named 'ConnectionRouter' in namespace 'noisepage::messenger'
common::ManagedPointer<messenger::ConnectionRouter> ListenAndMakeConnection(
                       ~~~~~~~~~~~^
../src/include/self_driving/model_server/model_server_manager.h:152:16: error: no type named 'CallbackFn' in namespace 'noisepage::messenger'
    messenger::CallbackFn model_server_logic);
    ~~~~~~~~~~~^
../src/include/self_driving/model_server/model_server_manager.h:267:48: error: no type named 'CallbackFn' in namespace 'noisepage::messenger'; did you mean 'Callback'?
  bool SendMessage(const std::string &payload, messenger::CallbackFn cb);
                                               ^~~~~~~~~~~~~~~~~~~~~
                                               Callback
../src/include/self_driving/model_server/model_server_manager.h:159:14: note: 'Callback' declared here
  enum class Callback : uint64_t { NOOP = 0, CONNECTED, DEFAULT };
             ^
../src/include/self_driving/model_server/model_server_manager.h:273:37: error: no member named 'ConnectionRouter' in namespace 'noisepage::messenger'
  common::ManagedPointer<messenger::ConnectionRouter> router_;
                         ~~~~~~~~~~~^
../src/include/self_driving/model_server/model_server_manager.h:276:8: error: no type named 'thread' in namespace 'std'
  std::thread thd_;
  ~~~~~^
In file included from CMakeFiles/noisepage_objlib.dir/Unity/unity_34_cxx.cxx:7:
/Users/ejeppinger/CLionProjects/terrier/src/self_driving/model_server/model_server_manager.cpp:129:26: error: out-of-line definition of 'SendMessage' does not match any declaration in 'noisepage::modelserver::ModelServerManager'
bool ModelServerManager::SendMessage(const std::string &payload, messenger::CallbackFn cb) {
                         ^~~~~~~~~~~
6 errors generated.
[8/34] Building CXX object CMakeFiles/noisepage_objlib.dir/Unity/unity_26_cxx.cxx.o

```

This comes from a diff:

```
diff --git a/src/self_driving/model_server/model_server_manager.cpp b/src/self_driving/model_server/model_server_manager.cpp
index d5b31de0..80bb4ce7 100644
--- a/src/self_driving/model_server/model_server_manager.cpp
+++ b/src/self_driving/model_server/model_server_manager.cpp
@@ -1,4 +1,7 @@
+#include "self_driving/model_server/model_server_manager.h"
+
 #include <sys/wait.h>
+
 #include <csignal>
 #include <thread>  // NOLINT

@@ -6,7 +9,6 @@
 #include "loggers/model_server_logger.h"
 #include "messenger/connection_destination.h"
 #include "messenger/messenger.h"
-#include "self_driving/model_server/model_server_manager.h"

 namespace noisepage::modelserver {
 static constexpr const char *MODEL_CONN_ID_NAME = "model-server-conn";
```

I'm not sure how that change could cause that compilation error, but ¯\_(ツ)_/¯

--- a/src/include/self_driving/model_server/model_server_manager.h
+++ b/src/include/self_driving/model_server/model_server_manager.h
@@ -12,14 +12,19 @@
 
 #pragma once
 
+#include <atomic>
+#include <condition_variable>
 #include <string>
+#include <thread>
 #include <utility>
 #include <vector>
 
 #include "common/json.h"
 #include "common/managed_pointer.h"
+#include "messenger/messenger_defs.h"
 
 namespace noisepage::messenger {
+class ConnectionRouter;
 class Messenger;
 }  // namespace noisepage::messenger
 

--- a/src/include/self_driving/model_server/model_server_manager.h
+++ b/src/include/self_driving/model_server/model_server_manager.h
@@ -13,9 +13,9 @@
 #pragma once
 
 #include <atomic>
-#include <condition_variable>
+#include <condition_variable>  // NOLINT
 #include <string>
-#include <thread>
+#include <thread>  // NOLINT
 #include <utility>
 #include <vector>
 

